components/authz-service/server/v2/project_update_manager.go

```diff
@@ -432,3 +404,45 @@ func (m *CerealProjectUpdateManager) Status() (ProjectUpdateStatus, error) {
 	}
 	return projectUpdateInstance, nil
 }
+
+// Find the longest estimate that is in the future. If no estimate is found return zero time.Time{}
+func findLongestEstimatedTime(estimatedTimes []time.Time) time.Time {
+	longestEstimatedTimeComplete := time.Time{}
+	for _, estimatedTime := range estimatedTimes {
+		if estimatedTime.After(time.Now()) && estimatedTime.After(longestEstimatedTimeComplete) {
+			longestEstimatedTimeComplete = estimatedTime
+		}
+	}
+
+	return longestEstimatedTimeComplete
+}
+
+func (w *workflowInstance) collectAllDomainServiceEstimates() []time.Time {
+	estimates := make([]time.Time, 0)
+	if !w.IsRunning() {
+		return estimates
+	}
+	domainServicesUpdateInstance, err := w.GetUpdateDomainServicesInstance()
+	if err == cereal.ErrWorkflowInstanceNotFound {
+		return estimates
+	}
+
+	for _, d := range domainServicesUpdateInstance.ListSubWorkflows() {
+		subWorkflow, err := domainServicesUpdateInstance.GetSubWorkflow(d)
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to get subworkflow for %q", d)
+			continue
+		}
+		payload := project_update_tags.DomainProjectUpdateWorkflowPayload{}
+		if subWorkflow.IsRunning() {
+			if err := subWorkflow.GetPayload(&payload); err != nil {
+				logrus.WithError(err).Errorf("failed to get payload for %q", d)
+				continue
+			}
+			estimatedTime := time.Unix(payload.MergedJobStatus.EstimatedEndTimeInSec, 0)
```

Comment:
For some reason I don't know about, we know that `payload.MergedJobStatus` is not a thing that could be `nil` and `EstimatedEndTimeInSec` is always available?

Reply:
That makes sense. I will add a check to see if it is nil.

Reply:
Wait,`payload.MergedJobStatus` can not be nil, but `payload.MergedJobStatus.EstimatedEndTimeInSec` could be zero.